### ISSUE TITLE
Wav2Vec custom data training / fine tuning outputs <unk> (possible repeat of #1112)

### ISSUE THREAD

**Issue by torphix** [PyTorchLightning/lightning-flash#1300]
## 🐛 Bug

Hi, I am using the code from the Wav2Vec fine tuning example (included below) on my custom dataset.
If I use the option     trainer.finetune(model, datamodule=datamodule, strategy='no_freeze') when I inference I get only <unk> tokens as predictions or blank strings. 
If i use  strategy='freeze' the model is also outputting <unk> tokens or blank strings
I have experimented with reducing learning rate to no avail.
Perhaps issue is related to #1112 ? however it seems to be addressed in PR #1275 
Do the wav files I am using need to be normalised / resampled or perhaps somethings else that I am not think of.
What is interesting is that like in #1112 using the non fine tuned base model results in normal output.

Thank you.

### To Reproduce

def finetune_flash(train_path, test_path):
    
    datamodule = SpeechRecognitionData.from_json(
        input_field="file",
        target_field="text",
        train_file=train_path,
        test_file=test_path,
        batch_size=4,
    )
    model = SpeechRecognition(backbone="facebook/wav2vec2-base-960h")
    trainer = flash.Trainer(max_epochs=5, gpus=1)
    trainer.finetune(model, datamodule=datamodule, strategy='no_freeze')
    # 4. Predict on audio files!
    datamodule = SpeechRecognitionData.from_files(predict_files=["data/timit/example.wav"], batch_size=4)
    predictions = trainer.predict(model, datamodule=datamodule)

    trainer.save_checkpoint('model.pth')    

    Predictions:  [['<unk><unk>']]

### Environment

 - OS (e.g., Linux): Linux
 - Python version: python==3.8
 - PyTorch/Lightning/Flash Version (e.g., 1.10/1.5/0.7):  lightning-flash==0.7.3
 - GPU models and configuration: Nvidia 2080ti
 - Any other relevant information:

**Comment by ethanwharris** [comment]
Hey @torphix thanks for reporting this! I would definitely recommend sticking with `strategy="freeze"`. You could try additionally lowering the learning rate to something like `learning_rate=1e-5` which may improve results. Another thing you could try is to use a learning rate schedule with warmup (e.g. following the doc here: https://lightning-flash.readthedocs.io/en/stable/general/optimization.html#schedulers)

It's also worth noting that the pretrained models are pretty good so they may work quite well straight out of the gate without any finetuning.

Hope that helps :smiley:

**Comment by torphix** [comment]
Thanks for getting back to me :)
I appreciate the feedback.
Whilst the pretrained models are fantastic I unfortunately need a higher accuracy for my task, also it would be great to get an understanding as to why this is occurring.
Interestingly the models only seem to do this when trained for a large number of steps so perhaps the model is just over fitting.

I understand if there is nothing further you can suggest and if you like I can go ahead and close the issue

Thanks again Ethan

**Comment by ethanwharris** [comment]
Hey @torphix, thanks for your comments!

> also it would be great to get an understanding as to why this is occurring

Yes, fully agree. As far as I can tell from my own experience it seems to relate to starting with a high loss which I guess makes the gradients go crazy and wipe out the pretrained weights (which causes trouble even when we freeze the backbone as the head is pretrained too).

I'll go ahead and close this issue as I unfortunately don't have any other suggestions, but please do report back if you find anything that works for your case. Long term it would be neat if we can include a section on tips and tricks for getting the most out of each flash task in our docs so anything you find out would be much appreciated 😃
